### PR TITLE
Chore: Update Action interface

### PR DESCRIPTION
**Pull request by adela-almasan**
This PR updates the `Action` interface to be go-sdk compatible.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Review by ryantxu — COMMENTED** [review]
LGTM -- I can add the go sdk side once this gets merged

**Review by ryantxu — APPROVED** [review]

**Review by leventebalogh — COMMENTED** [review]
Although I couldn't find any plugins depending on the changed properties, technically this could be a breaking change.

**Review by baldm0mma — APPROVED** [review]

**Comment by adela-almasan** [comment]
Added Canvas migration to go around the breaking changes, although I'm not sure if the feature is used yet as it was just merged and it's under a feature flag. @ryantxu when you have a minute, do you mind updating the sdk?

**Pull request merged**
